### PR TITLE
fix(discover): Allow custom performance metrics to be used in discover arithmetic queries

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
Allows custom performance metrics to be used in discover arithmetic queries.

### CODE REVIEW

File: src/sentry/search/events/builder.py [hunk 1]
```diff
@@ -868,6 +869,9 @@ def custom_measurement_map(self) -> List[MetricMeta]:
             return []
         return result
 
+    def get_custom_measurement_names_set(self) -> Set[str]:
+        return set(map(lambda measurement: measurement["name"], self.custom_measurement_map))
```
Comment (wmak) [inline src/sentry/search/events/builder.py:873]: nit:
```suggestion
        return {measurement["name"] for measurement in self.custom_measurement_map}
```

**Review by wmak — APPROVED** [review]

**Review by k-fish — APPROVED** [review]
👍 but only if you ignore Will's nit 😆

**Comment by wmak** [comment]
okok i have a better suggestion @k-fish
```
return set(map(lambda measurement: measurement["name"], measurement["name"] for measurement in self.custom_measurement_map}))
````
just do both 😎

**Comment by k-fish** [comment]
@wmak okay actually that's a great idea, very high quality 👍

**Comment by k-fish** [comment]
@edwardgou-sentry 😠

**Pull request merged**